test/unit-tests/testAnalyticsFramework.js

```diff
@@ -569,19 +570,79 @@ describe('Analytics Framework Unit Tests', function()
     var badParamsHelper = function(framework, params, msgSentObj)
     {
       var msgName;
-      for(msgName in OO.Analytics.EVENTS)
+      var events = framework.flattenEvents(OO.Analytics.EVENTS); 
+      for(var i = 0; i < events.length; i++)
       {
-        expect(framework.publishEvent(OO.Analytics.EVENTS[msgName])).toBe(true);
+        msgName = events[i];
+        expect(framework.publishEvent(msgName)).toBe(true);
         msgSentObj.count++;
         recordedEvents = framework.getRecordedEvents();
         expect(_.isArray(recordedEvents)).toBe(true);
         var length = recordedEvents.length;
         expect(length).toEqual(msgSentObj.count);
         var lastMsg = recordedEvents[length-1];
-        expect(lastMsg.eventName).toEqual(OO.Analytics.EVENTS[msgName]);
+        expect(lastMsg.eventName).toEqual(msgName);
       }
     };
 
+    describe('Test Framework Initialization', function()
+    {
+      it('Test Flatten Events Function', function()
+      {
+        // Should return the correct number of values
+        var object = {};
+        expect(framework.flattenEvents(object).length).toBe(0);
+
+        object = {
+          a: true,
+          b: true,
+          c: true
+        };
+        expect(framework.flattenEvents(object).length).toBe(3);
+
+        object = {
+          a: {
+            aa1: true,
+            aa2: true,
+            aa3: true
+          },
+          b: {
+            bb: true
+          }
+        };
+        expect(framework.flattenEvents(object).length).toBe(4);
+
+        object = {
+          a: {
+            aa1: true,
+            aa2: true,
+          },
+          b: {},
+          c: true,
+          d: true,
+          e: true
+        };
+        expect(framework.flattenEvents(object).length).toBe(5);
+
+        object = {
+          a: {
+            aa1: true,
+            aa2: {
+              aaa1: true
+            },
+          },
+          b: {
+            b1: {
+              bb1: {
+                bbb1: true
+              }
+            }
+          }
+        };
+        expect(framework.flattenEvents(object).length).toBe(3);
+      });
+    });
```

Comment:
you should also double check that the values are what you expect. Not just the count.